drivers/SmartThings/zigbee-switch/src/multi-switch-no-master/init.lua

```diff
@@ -11,61 +11,118 @@
-  { model = "E220-KR4N0Z0-HA" },
-  { model = "E220-KR5N0Z0-HA" },
-  { model = "E220-KR6N0Z0-HA" }
+  { mfr = "DAWON_DNS", model = "PM-S240-ZB", children = 1 },
+  { mfr = "DAWON_DNS", model = "PM-S240R-ZB", children = 1 },
+  { mfr = "DAWON_DNS", model = "PM-S250-ZB", children = 1 },
+  { mfr = "DAWON_DNS", model = "PM-S340-ZB", children = 2 },
+  { mfr = "DAWON_DNS", model = "PM-S340R-ZB", children = 2 },
+  { mfr = "DAWON_DNS", model = "PM-S350-ZB", children = 2 },
+  { mfr = "DAWON_DNS", model = "ST-S250-ZB", children = 1 },
+  { mfr = "DAWON_DNS", model = "ST-S350-ZB", children = 2 },
+  { mfr = "ORVIBO", model = "074b3ffba5a045b7afd94c47079dd553", children = 2 },
+  { mfr = "ORVIBO", model = "9f76c9f31b4c4a499e3aca0977ac4494", children = 3 },
+  { mfr = "REXENSE", model = "HY0002", children = 2 },
+  { mfr = "REXENSE", model = "HY0003", children = 3 },
+  { mfr = "REX", model = "HY0096", children = 2 },
+  { mfr = "REX", model = "HY0097", children = 3 },
+  { mfr = "HEIMAN", model = "HS2SW2L-EFR-3.0", children = 2 },
+  { mfr = "HEIMAN", model = "HS2SW3L-EFR-3.0", children = 3 },
+  { mfr = "eWeLink", model = "ZB-SW02", children = 2 },
+  { mfr = "eWeLink", model = "ZB-SW03", children = 3 },
+  { mfr = "eWeLink", model = "ZB-SW04", children = 4 },
+  { model = "E220-KR2N0Z0-HA", children = 1 },
+  { model = "E220-KR3N0Z0-HA", children = 2 },
+  { model = "E220-KR4N0Z0-HA", children = 3 },
+  { model = "E220-KR5N0Z0-HA", children = 4 },
+  { model = "E220-KR6N0Z0-HA", children = 5 }
 }
 
 local function is_multi_switch_no_master(opts, driver, device)
   for _, fingerprint in ipairs(MULTI_SWITCH_NO_MASTER_FINGERPRINTS) do
-      if device:get_manufacturer() == nil and device:get_model() == fingerprint.model then
-        return true
-      elseif device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
-        return true
-      end
+    if device:get_manufacturer() == nil and device:get_model() == fingerprint.model then
+      return true
+    elseif device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
   end
-
   return false
 end
 
-local function component_to_endpoint(device, component_id)
-  local ep_num = component_id:match("switch(%d)")
-  return ep_num and tonumber(ep_num) + 1 or 1
+local function get_children_amount(device)
+  for _, fingerprint in ipairs(MULTI_SWITCH_NO_MASTER_FINGERPRINTS) do
+    if device:get_model() == fingerprint.model then
+      return fingerprint.children
+    end
+  end
 end
 
-local function endpoint_to_component(device, ep)
-  local switch_comp = string.format("switch%d", ep - 1)
-  if device.profile.components[switch_comp] ~= nil then
-    return switch_comp
+local function do_refresh(self, device)
+  local children_amount = get_children_amount(device)
+  for endpoint = PARENT_ENDPOINT, children_amount+1, 1 do
+    device:send(OnOff.attributes.OnOff:read(device):to_endpoint(endpoint))
+  end
+end
```

Comment:
I meant to refresh a particular endpoint here.

Reply:
I thougth that this implementation refreshes a particular device: